pages/api/mail/index.ts

```diff
@@ -0,0 +1,111 @@
+        // stores all promises for the nodemailer transport
+        const mailerPromises = [];
+        // looping through all the classes starting in three hours
+        for (let i = 0; i < classesInThreeHours.length; ++i) {
+            // looping through parents that are registered to EACH class
+            for (let j = 0; j < classesInThreeHours[i].parentRegs.length; ++j) {
+                mailerPromises.push(
+                    send(
+                        process.env.EMAIL_FROM,
+                        classesInThreeHours[i].parentRegs[j].parent.user.email,
+                        "Reminder: Social Diversity for Children Class In 3 Hours",
+                        `<p>Hi ${classesInThreeHours[i].parentRegs[j].parent.user.firstName},</p>
+                        <p>The class <b>${classesInThreeHours[i].name}</b> you signed up for is starting in 3 hours!</p>
+                        <p>Regards, Social Diversity for Children</p>`,
+                    ),
+                );
+            }
+            // looping through volunteers that are registered to EACH class
+            for (
+                let j = 0;
+                j < classesInThreeHours[i].volunteerRegs.length;
+                ++j
+            ) {
+                mailerPromises.push(
+                    send(
+                        process.env.EMAIL_FROM,
+                        classesInThreeHours[i].volunteerRegs[j].volunteer.user
+                            .email,
+                        "Reminder: Social Diversity for Children Class In 3 Hours",
+                        `<p>Hi ${classesInThreeHours[i].volunteerRegs[j].volunteer.user.firstName},</p>
+                        <p>The class ${classesInThreeHours[i].name} you signed up for is starting in 3 hours!</p><br />
+                        <p>Regards, Social Diversity for Children</p>`,
+                    ),
+                );
+            }
+        }
+        // looping through all the classes starting in forty-eight hours
+        for (let i = 0; i < classesInFortyEightHours.length; ++i) {
+            // looping through parents that are registered to EACH class
+            for (
+                let j = 0;
+                j < classesInFortyEightHours[i].parentRegs.length;
+                ++j
+            ) {
+                mailerPromises.push(
+                    send(
+                        process.env.EMAIL_FROM,
+                        classesInThreeHours[i].parentRegs[j].parent.user.email,
+                        "Reminder: Social Diversity for Children Class In 48 Hours",
+                        `<p>Hi ${classesInThreeHours[i].parentRegs[j].parent.user.firstName},</p>
+                        <p>The class ${classesInThreeHours[i].name} you signed up for is starting in 48 hours!</p><br />
+                        <p>Regards, Social Diversity for Children</p>`,
+                    ),
+                );
+            }
```

Comment:
this exact block seems to be repeated a few times - can we make a helper function with parameters passed in?